File: Day3/solp1.py
import re

def calculate_mul_sum(text):
    pattern = r'mul\((\d{1,3}),(\d{1,3})\)'
    matches = re.finditer(pattern, text)
    
    total = 0
    for match in matches:
        x = int(match.group(1))
        y = int(match.group(2))
        total += x * y
    
    return total

# Read and process the input file
total_sum = 0
with open('inputs.txt', 'r') as file:
    for line in file:
        if line.strip():
            line_sum = calculate_mul_sum(line)
            total_sum += line_sum

print(f"Sum of all multiplication results: {total_sum}")

File: Day3/solp1.swift
import Foundation

func calculateMulSum(_ text: String) -> Int {
    let pattern = "mul\\((\\d{1,3}),(\\d{1,3})\\)"
    var total = 0

    do {
        let regex = try NSRegularExpression(pattern: pattern, options: [])
        let range = NSRange(text.startIndex..., in: text)

        let matches = regex.matches(in: text, range: range)

        for match in matches {
            if let xRange = Range(match.range(at: 1), in: text),
                let yRange = Range(match.range(at: 2), in: text),
                let x = Int(text[xRange]),
                let y = Int(text[yRange])
            {
                total += x * y
            }
        }
    } catch {
        print("Regex error: \(error)")
    }

    return total
}

// Get the current file's directory
let currentFilePath = #file
let currentDirectory = URL(fileURLWithPath: currentFilePath).deletingLastPathComponent().path

// Construct path to inputs.txt in the same directory
let inputPath = currentDirectory + "/inputs.txt"

if let fileContents = try? String(contentsOfFile: inputPath, encoding: .utf8) {
    var totalSum = 0
    let lines = fileContents.components(separatedBy: .newlines)

    for line in lines where !line.isEmpty {
        let lineSum = calculateMulSum(line)
        totalSum += lineSum
    }

    print("Sum of all multiplication results: \(totalSum)")
} else {
    print("Error reading file")
}
